framework/src/meshgenerators/TransitionLayerConnector.C

```diff
@@ -0,0 +1,205 @@
+                                    "Boundary ID to be assigned to the boundary connecting ending "
+                                    "points of the positions_vectors.");
+  params.addParam<bool>(
+      "use_quad_elements",
+      false,
+      "Whether QUAD4 instead of TRI3 elements are used to construct the transition layer.");
+  params.addRangeCheckedParam<Real>(
+      "bias_parameter",
+      1.0,
+      "bias_parameter>=0",
+      "Parameter used to set up biasing of the layers: bias_parameter > 0.0 is used as the biasing "
+      "factor; bias_parameter = 0.0 activates automatic biasing based on local node density on "
+      "both input boundaries.");
+  params.addRangeCheckedParam<Real>(
+      "gaussian_sigma",
+      3.0,
+      "gaussian_sigma>0.0",
+      "Gaussian parameter used to smoothen local node density for automatic biasing; this "
+      "parameter is not used if another biasing option is selected.");
+  params.addParam<bool>(
+      "keep_inputs",
+      true,
+      "Whether to output the input meshes stitched with the transition layer connector.");
+  params.addClassDescription("This TransitionLayerConnector object is designed to generate a "
+                             "transition layer to connect two boundaries of two input meshes.");
+  return params;
+}
+
+TransitionLayerConnector::TransitionLayerConnector(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input_name_1(getParam<MeshGeneratorName>("input_mesh_1")),
+    _input_name_2(getParam<MeshGeneratorName>("input_mesh_2")),
+    _boundary_1(getParam<std::vector<BoundaryName>>("boundary_1")),
+    _boundary_2(getParam<std::vector<BoundaryName>>("boundary_2")),
+    _mesh_1_shift(getParam<Point>("mesh_1_shift")),
+    _mesh_2_shift(getParam<Point>("mesh_2_shift")),
+    _num_layers(getParam<unsigned int>("num_layers")),
+    _block_id(getParam<subdomain_id_type>("block_id")),
+    _input_boundary_1_id(getParam<boundary_id_type>("input_boundary_1_id")),
+    _input_boundary_2_id(getParam<boundary_id_type>("input_boundary_2_id")),
+    _begin_side_boundary_id(getParam<boundary_id_type>("begin_side_boundary_id")),
+    _end_side_boundary_id(getParam<boundary_id_type>("end_side_boundary_id")),
+    _use_quad_elements(getParam<bool>("use_quad_elements")),
+    _bias_parameter(getParam<Real>("bias_parameter")),
+    _sigma(getParam<Real>("gaussian_sigma")),
+    _keep_inputs(getParam<bool>("keep_inputs")),
+    _input_1(getMeshByName(_input_name_1)),
+    _input_2(getMeshByName(_input_name_2))
+{
+}
+
+std::unique_ptr<MeshBase>
+TransitionLayerConnector::generate()
+{
+  auto input_mesh_1 = dynamic_cast<ReplicatedMesh *>(_input_1.get());
+  auto input_mesh_2 = dynamic_cast<ReplicatedMesh *>(_input_2.get());
+  if (!input_mesh_1)
+    paramError("input_mesh_1", "Input is not a replicated mesh, which is required.");
+  if (*(input_mesh_1->elem_dimensions().begin()) != 2 ||
+      *(input_mesh_1->elem_dimensions().rbegin()) != 2)
```

Comment:
why not use mesh_i->mesh_dimension() ?

Reply:
`mesh_dimension()` is the maximum of `elem_dimensions()`. I used the current one to make sure that all the elements are 2D.